mk/footer.mk

```diff
@@ -0,0 +1,2 @@
+d := $(dirstack_$(sp))
+sp := $(basename $(sp))
```

Comment:
it keeps track of directory stack so when one Rules.mk in sub directory exists, the $(d) is restored to previous one in stack. It is standard non-recursive Makefile boilerplate.